moveit_ros/moveit_servo/demos/cpp_interface/demo_joint_jog.cpp

```diff
@@ -0,0 +1,101 @@
+ *   and/or other materials provided with the distribution.
+ *
+ * * Neither the name of the copyright holder nor the names of its
+ *   contributors may be used to endorse or promote products derived from
+ *   this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *******************************************************************************/
+
+/*      Title     : joint_jog_demo.cpp
+ *      Project   : moveit_servo
+ *      Created   : 27/05/2023
+ *      Author    : V Mohammed Ibrahim
+ */
+
+#include <iostream>
+#include <chrono>
+#include <rclcpp/rclcpp.hpp>
+#include <tf2_eigen/tf2_eigen.hpp>
+#include <moveit_servo/servo.hpp>
+#include <moveit_servo/utils.hpp>
+
+using namespace moveit_servo;
+
+const rclcpp::Logger LOGGER = rclcpp::get_logger("moveit_servo.joint_jog_demo");
+
+int main(int argc, char* argv[])
+{
+  rclcpp::init(argc, argv);
+
+  // The servo object expects to get a ROS node.
+  auto servo_node = std::make_shared<rclcpp::Node>("servo_node");
+
+  // Get the servo parameters.
+  std::string param_namespace = "moveit_servo";
+  auto servo_param_listener = std::make_shared<const servo::ParamListener>(servo_node, param_namespace);
+  auto servo_params = servo_param_listener->get_params();
+
+  // The publisher to send trajectory message to the robot controller.
+  rclcpp::Publisher<trajectory_msgs::msg::JointTrajectory>::SharedPtr trajectory_outgoing_cmd_pub;
+  trajectory_outgoing_cmd_pub = servo_node->create_publisher<trajectory_msgs::msg::JointTrajectory>(
+      servo_params.command_out_topic, rclcpp::SystemDefaultsQoS());
+
+  // Create the servo object
+  auto servo = Servo(servo_node, servo_param_listener);
+
+  // Wait for some time, so that we can actually see when the robot moves.
+  // This is just for convenience, should not be used for sync in real application.
+  std::this_thread::sleep_for(std::chrono::seconds(3));
+
+  // Frquency at which the commands will be send to robot controller.
+  rclcpp::WallRate rate(1.0 / servo_params.publish_period);
+
+  // Set the command type for servo.
+  servo.incomingCommandType(CommandType::JOINT_JOG);
+
+  RCLCPP_INFO_STREAM(LOGGER, servo.getStatusMessage());
```

Comment:
Seems like this line doesn't do anything?

```suggestion
```

Reply:
This is here just for convenience to print the status before beginning the demo.